src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -1283,5 +1290,65 @@ public async Task DeletePullRequestBranchAsync(string pullRequestUri)
             (string owner, string repo, int id) prInfo = ParsePullRequestUri(pullRequestUri);
             await DeleteBranchAsync(prInfo.owner, prInfo.repo, pr.HeadBranch);
         }
+
+        /// <summary>
+        /// Log extra details from ApiException.
+        /// 
+        /// GitHub returns a GitHub-custom JSON payload on most errors that contains more information than
+        /// the standard exception message. See <see href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#client-errors">Client Errors</see> 
+        /// for details on the structure.
+        /// </summary>
+        private void LogApiErrorDetails(ApiException apiException)
+        {
+            _logger.LogError(apiException, "Caught ApiException, will rethrow -- StatusCode: {StatusCode}", apiException.StatusCode);
+
+            if (!string.IsNullOrEmpty(apiException.ApiError.DocumentationUrl))
+            {
+                _logger.LogError("Documentation URL: {DocumentationUrl}", apiException.ApiError.DocumentationUrl);
```

Comment:
These should probably be just be details of the original log. It can be hard to correlate 3 error logs together, since they can get interspersed with many others.

Reply:
This method also logs all "error" entries from the response, so even more lines. 

https://github.com/dotnet/arcade-services/blob/6e4f7ff49dbe81a3b568ab56c96da4249b2fa9ba/src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs#L1310-L1315

My thought was that the operation IDs are enough to isolate instances. 

Open to other ideas, of course. I felt that just using a single log message would be harder to make sense of.

Reply:
I seem to remember a single operation id didn't correlate all that tightly...

If we expect that there will be not too many error details typically? < 2 perhaps? then I agree with @ChadNedzlek  that including these in a single message will be better.

Reply:
Fair. 

How would you feel if I serialized it all to JSON? It's weird because it's the only message like that. It's good because Kusto could make sense of it.

Reply:
Just add them as structured logging values.  Then you can get them with "customDimention.DocumentationUrl" and such.  Unfortunately, it would be best to just have them as properties on the trace, but ILogger doesn't expose that. Basically ALL these logs are just adding additional properties, the message itself isn't really all that useful, so I'd just have the message be `"StatusCode: {StatusCode}, DocumentaionUrl: {DocumentationUrl}, Code: {Code}"` and such.

Reply:
I'm struggling a bit with this feedback. Making a single log message makes sense, no problem there. But the entirety of the information I'm trying to capture is too complex for an ad-hoc format string. To me it just makes a lot of sense to borrow JSON as a way to structure the data. And as an extra win, it will handle however few or many items actually end up in the details list. The JSON itself will appear in the structured logging fields, so super easy to parse out of the log. The only risk that I see is that throwing the serializer in at this point in the program crashing could be considered expensive, but... it's not really? 

Happy to consider more here but I'll need some help understanding alternatives.

Reply:
The "DocumentationUrl" can just go in the first message. Let it be null/empty if it's null/empty.  The others, being a list per exception, probably isn't feasible to put inline, having them as a JSON blob works well enough. :-)

Reply:
Ah, okay, I see where we are now. Yeah, that list is actually the most interesting piece of info to capture. The other fields are good but the list is where the action is.

Reply:
I reformatted it a bit. Let me know if you approve (or better, let GitHub know)